server/src/main/kotlin/suwayomi/tachidesk/manga/controller/UpdateController.kt

```diff
@@ -93,11 +93,14 @@ object UpdateController {
         if (clear) {
             updater.reset()
         }
+
+        val categoriesToExclude = categories.filter { !it.includeInUpdate }
         categories
+            .filter { it.includeInUpdate }
             .flatMap { CategoryManga.getCategoryMangaList(it.id) }
             .distinctBy { it.id }
             .sortedWith(compareBy(String.CASE_INSENSITIVE_ORDER, MangaDataClass::title))
-            .filter { it.updateStrategy == UpdateStrategy.ALWAYS_UPDATE }
+            .filter { it.updateStrategy == UpdateStrategy.ALWAYS_UPDATE && !categoriesToExclude.any { category -> CategoryManga.getMangaCategories(it.id).contains(category) } }
```

Comment:
This will create a massive amount of queries to the database, there would probably be a better way.

Reply:
good point, didn't think about that

changed it to get the categories for all mangas of the categories that should get updated in one query (not sure if that can be optimized, my sql knowledge is rotting away and it took me way too long to get it working with exposed 😭)